module/VuFind/src/VuFind/Recommend/InjectSearchRunnerFactory.php

```diff
@@ -70,6 +70,10 @@ public function __invoke(
         if (!empty($options)) {
             throw new \Exception('Unexpected options passed to factory.');
         }
-        return new $requestedName($container->get(SearchRunner::class));
+        $module = new $requestedName($container->get(SearchRunner::class));
+        if (method_exists($module, 'setConfigManager')) {
```

Comment:
First off, there may be a better way to make the config manager available within the recommendation module.  Even if not, `method_exists` is ugly but I'm not sure what is best.  Maybe an interface that I check with instanceof, and define the setter in a trait?  But that seems overkill and I couldn't come up with a middle ground.

Reply:
Maybe the solution is just to create a different factory. The InjectSearchRunnerFactory is designed to do exactly what it says: inject the SearchRunner into the constructor. But now you want to inject the SearchRunner AND set the config manager. Maybe the easiest solution is to create an AbstractSearchObjectFactory -- then you can do all the injections you want safely since the factory is tailored for AbstractSearchObject and can assume that's how it will be used.

(I think the only reason we have InjectSearchRunnerFactory is because the various Results modules didn't inherit from a common base until you refactored them, and we probably didn't think to change the factory at that time).

In any case, if you agree with this approach, we can change all the Results modules and see if anything else is still using InjectSearchRunnerFactory. If not, we can probably deprecate it and remove it in 11.0.

Reply:
This makes sense; implemented.  And yes InjectSearchRunnerFactory is only used for those specific modules.